master/internal/experiment.go

```diff
@@ -633,6 +639,29 @@ func (e *experiment) setOrder(ctx *actor.Context, queuePosition float64) error {
 	return nil
 }
 
+func (e *experiment) setRP(ctx *actor.Context, msg job.SetRP) error {
+	if sproto.UseK8sRM(ctx.Self().System()) {
+		ctx.Respond(fmt.Errorf("kubernetes does not support setting resource pools"))
+	}
+
+	if _, err := sproto.GetResourcePool(ctx.Self().System(), msg.ResourcePool, 0, false); err != nil {
+		ctx.Respond(fmt.Errorf("invalid resource pool name"))
+		return nil
+	}
+
+	resources := e.Config.Resources()
+	resources.SetResourcePool(msg.ResourcePool)
+	e.Config.SetResources(resources)
+
+	if err := e.db.SaveExperimentConfig(e.Experiment); err != nil {
+		return errors.Wrapf(err, "setting experiment %s RP to %s", e.ID, msg.ResourcePool)
+	}
+	for _, ref := range ctx.Children() {
```

Comment:
I think we have an actor method for sending the same message to a list of refs. that might help us wait for the responses on all of them as well before returning

Reply:
somewhat fixed - I think each error caught here is worthy of being stopped for on its own.

Reply:
hmm I'm not sure I understand. what sort of error you're seeing this catch?
I was thinking about `askAll` or `getAll` I'm not sure how they handled error responses